core/src/main/java/org/jdbi/v3/core/StatementContext.java

```diff
@@ -155,6 +157,13 @@ public Object getAttribute(String key)
         return config.collectorRegistry.findCollectorFor(type);
     }
 
+    public <C extends ExtensionConfig<C>> C findConfiguration(Class<C> configType)
```

Comment:
`configType` might be an interface, in which case it's possible there are multiple registered config types that match it.

I have to think about this for a bit.